daemon/graphdriver/overlay2/overlay.go

```diff
@@ -174,6 +175,14 @@ func Init(home string, options []string, uidMaps, gidMaps []idtools.IDMap) (grap
 		}
 	}
 
+	// If kernel is older than 4.8, check whether opaque copy up bug has been patched
+	if kernel.CompareKernelVersion(*v, kernel.VersionInfo{Kernel: 4, Major: 8, Minor: 0}) < 0 {
```

Comment:
Why should we skip the kernel check though? The fix was merged before 4.8 was released so we can be confident that 4.8 or later have the fixed. Kernel tests were also added around this issue to prevent regression in later versions. Ideally users on newer Kernels won't have to incur the additional startup time required for this check.